doc/source/usage.rst

```diff
@@ -0,0 +1,194 @@
+
+Model mappings, defined on a per-model basis serve three different purposes:
+
+1. Define a list of model native regions that are to be selected (and
+   usually uploaded) from an IAM result. This also serves as an implicit
+   exclusion list for model native regions, since only explicitly
+   mentioned regions are selected.
+
+2. Allow for renaming of model native regions.
+
+3. Define how model native regions should be aggregated to common
+   regions.
+
+This example illustrates how such a model mapping looks like:
+
+.. code:: yaml
+
+   model: model_a
+   native_regions:
+     - region_a: alternative_name_a
+     - region_b
+   common_regions:
+     - common_region_1:
+       - region_a
+       - region_b
+     - common_region_2:
+       - ...
+
+**Notes**
+
+-  The names of the three top level keywords **model**,
+   **native_regions** and **common_regions** are fixed.
+-  Required properties are **model** and **at least** either
+   **native_regions** or **common_regions**. **Both** are **allowed** as
+   well.
+-  **model** (str): specifies the model name for which the mapping
+   applies.
+-  **native_regions** (list): list of model native regions serves as
+   a selection as to which regions to keep.
+
+   -  In the above example *region_a* is to be renamed to
+      *alternative_name_a*. This is done by defining a key-value pair
+      of *model_native_name: new_name*.
+   -  *region_b* is selected but the name is not changed.
+   -  Assuming *model_a* also defines a third region *region_c*,
+      since it is not mentioned it will be **dropped** from the data.
+
+-  **common_regions** (list): list of common regions which will be
+   computed as aggregates. They are defined as list entries which
+   themselves have a list of constituent regions. These constituent
+   regions must be model native regions.
+
+   -  **Important to note** the names of the constituent regions
+      **must** refer to the **original** model native region names. In
+      the above example *region_a* and *region_b* and **not**
+      *alternative_name_a*.
+
+Guidelines and variable naming conventions
+------------------------------------------
+
+The variable name (code) should adhere to the following conventions:
+
+-  A *|* (pipe) character indicates levels of hierarchy
+-  Do not use spaces before and after the *|* character, but add a
+   space between words (e.g., *Primary Energy|Non-Biomass Renewables*)
+-  All words must be capitalised (except for *and*, *w/*, *w/o*, etc.)
+-  Do not use abbreviations (e.g, *PHEV*) unless strictly necessary
+-  Add hierarchy levels where it might be useful in the future, e.g.,
+   use *Electric Vehicle|Plugin-Hybrid* instead of *Plugin-Hybrid
+   Electric Vehicle*
+-  Do not use abbreviations of statistical operations (*min*, *max*,
+   *avg*) but always spell out the word
+-  Do not include words like *Level* or *Quantity* in the variable,
+   because this should be clear from the context or unit
+
+Class Overview
```

Comment:
This section doesn't really make sense at the bottom. Best to integrate the descriptions into the text above.

Reply:
Will do, should I start using the linked references for every mention of a class like you started in #77?
So :class:`nomenclature.codelist.Codelist` instead of **CodeLists**?

Reply:
Yes, we should use class-references wherever possible (that's why I started #77 today, to allow adopting this consistently right away).

Reply:
Done